tracing-opentelemetry/Cargo.toml

```diff
@@ -31,11 +31,15 @@ tracing-core = { path = "../tracing-core", version = "0.1.28" }
 tracing-subscriber = { path = "../tracing-subscriber", version = "0.3.0", default-features = false, features = ["registry", "std"] }
 tracing-log = { path = "../tracing-log", version = "0.1.3", default-features = false, optional = true }
 once_cell = "1.13.0"
+instant = "0.1.12"
 
 # Fix minimal-versions
 async-trait = { version = "0.1.56", optional = true }
 thiserror = { version = "1.0.31", optional = true }
 
+[target.wasm32-unknown-unknown.dependencies]
```

Comment:
i'm not all that familiar with the details of compiling Rust to WebAssembly, so, just to clarify: is the dependency on `js-sys` necessary only for `wasm32-unknown-unknown`, and _not_ for `wasm32-wasi` targets? does `std::time::SystemTime` work on `wasm32-wasi`?

Reply:
Unclear. When I originally patched this (months ago), I simply copied the entire approach from https://github.com/open-telemetry/opentelemetry-rust as far as I can remember. Looking at it now, their [dependency](https://github.com/open-telemetry/opentelemetry-rust/blob/9ea1ee77af8577264ddc21e9f1e009aa38437bf8/opentelemetry-api/Cargo.toml#L22-L23) currently matches the cfg in the code: `target_arch = "wasm32"`

Which looks a bit more reasonable given that the code I committed would certainly not work on `wasm32-wasi`, good catch. On the other hand, I have never worked with `wasm32-wasi`, so I honestly have no idea whether `SystemTime` or `js_sys::Date` works there. A quick google search also didn't give me any meaningful results and unfortunately I don't have the bandwidth right now to dig deeper into this. Hence, I would propose to just gate both the dependency and the `now()` implementation to `wasm32-unknown-unknown` to avoid accidentally breaking anything `wasm32-wasi`.

Reply:
hmm, if `opentelemetry-api` uses `js-sys` for all wasm targets, perhaps we should as well? it seems like we wouldn't want to break support on some wasm platforms that opentelemetry works with. but, more investigation is probably good.

Reply:
I'd investigate further, support in `opentelemetry-api` was fairly experimental as I recall